Mathport/Syntax/Translate/Basic.lean

```diff
@@ -956,11 +957,9 @@ def trExpr' : Expr → M Syntax
         `(Parser.Term.structInstField| $lhsId:ident := $(← trExpr rhs))
     -- TODO(Mario): formatter has trouble if you omit the commas
     if catchall then
-      `({ $[$srcs,* with]? $[$flds:structInstField, ]* .. })
-    else if let some last := flds.back? then
-      `({ $[$srcs,* with]? $[$(flds.pop):structInstField, ]* $last:structInstField })
+      `({ $[$srcs,* with]? $[$flds:structInstField],* .. })
```

Comment:
Can the TODO be addressed now? I originally wanted to write this expression without any commas at all, but the formatter would produce ill formed output in that case.

Reply:
No, Sebastian's PR only added the sepByIdent parser combinatorial.  With this we can now easily write a proper formatter that prints the correct newlines (and omits them if all separators are present).  But that still needs to be done.

Reply:
You would have to replace the `SepArray.ofElems` coercion to `SepArray ","` happening here. But as I said in the lean4 PR, ideally the pretty printer should be the one choosing the sensible separator.

Reply:
To add to Sebastian's comment.  Right now, `$fields,*` always adds the comma, with no good way to opt out.  Maybe we could add `$fields*` for the commaless version?

Longer term I agree, the formatter (or some other part of the pretty printer) should be able to choose or modify the separators.

Reply:
> Right now, $fields,* always adds the comma, with no good way to opt out

Just to be clear, it's not the splice itself that adds the comma.
```
let flds := ⟨mkSepArray flds mkNullNode⟩
```
should change it to line breaks, I think

Reply:
The `mkSepArray` is a nice trick to change the separators.  We still need to fix the formatter though.

Reply:
> ideally the pretty printer should be the one choosing the sensible separator.

Note that this is not entirely trivial, since we can't just skip tokens.  There might be a comment associated to the comma after all.

In the meantime, a comma-less antiquotation would be really helpful.  Particularly in more complicated cases like `def foo : Foo where $[$names := $values]*` where it's hard to insert the `mkSepArray` (e.g. because the child parser of the sepByIndent doesn't have a name or antiquotation).  Would that be hard to add, the antiquotations are a bit too magic to me?